meow/frontend/src/components/EditPost/EditForm.js

```diff
@@ -119,7 +120,20 @@ class EditForm extends React.Component {
     setFieldsValue({ tags });
   };
 
+  handleClickpub = e => {
+    this.setState({
+      ...this.state,
+      click_pubready: e.target.checked
+    });
+  };
+
   render() {
+    let flag_pub = null;
+    if (this.state.click_pubready && (this.props.story_url == "" || this.props.story_url == null)) {
```

Comment:
You can actually get this value from some form helper functions. Ant design (which is the UI framework we use) has a very complex form component that we use here. Here's the documentation to [it](https://ant.design/components/form/). There is a helper function for getting a value inside the form. 

I would recommend using that function instead because it reduces redundancy. Have two variables which store the same thing (in this case `click_pubready` and ant design form internal state for the checkbox) is not good because its confusing to other readers and can lead to weird bugs if states become different some how.
I think this code show work.
```
    const { getFieldDecorator, getFieldsValue } = this.props.form; // this line already exists in code, you'll just need to move it so you can use tjhe variables it declares when you need it.

        console.log(getFieldsValue(["name of field"]));
```
You can find the name of the field by looking at this line

```
{getFieldDecorator("pub_ready_copy", {
            rules: [],
            valuePropName: "checked",
            initialValue: false
          })(<Checkbox style={{ fontSize: "1.2em" }}>Copy-edited</Checkbox>)}
```

The `getFieldDecorator(name, options, input )` function will sorta register that input component to have the field name passed into the function.